### PR TITLE
db/insert-many! -> toucan2

### PR DESCRIPTION
**Pull request by qnkhuat**
Epic: #29231

the equivalent of `(db/insert-many! modelable row)` in toucan 2 is `(t2/insert-returning-pks! modelable row)`

Since both `insert-many` and `t2/insert-returning-pks` do trigger `post-insert` so we don't have to worry about behavior changes here.

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/29510)

### CODE REVIEW

File: src/metabase/models/permissions.clj [hunk 1]
```diff
@@ -1037,10 +1036,10 @@
    (t2/delete! Permissions :group_id (u/the-id group-or-id) :object [:like "/query/%"])
    (t2/delete! Permissions :group_id (u/the-id group-or-id) :object [:like "/data/%"])
    (try
-     (db/insert-many! Permissions
-       (map (fn [path-object]
-              {:group_id (u/the-id group-or-id) :object path-object})
-            (distinct (conj (->v2-path path) path))))
+     (t2/insert! Permissions
```
Comment (tsmacdonald) [inline src/metabase/models/permissions.clj:1039]: shouldn't this be an `insert-returning-instances!`?

File: src/metabase/sync/sync_metadata/fields/sync_instances.clj [hunk 2]
```diff
@@ -41,7 +41,7 @@
   "Insert new Field rows for for all the Fields described by `new-field-metadatas`."
   [table :- i/TableInstance, new-field-metadatas :- [i/TableMetadataField], parent-id :- common/ParentID]
   (when (seq new-field-metadatas)
-    (db/insert-many! Field
+    (t2/insert-returning-pks! Field
```
Comment (tsmacdonald) [inline src/metabase/sync/sync_metadata/fields/sync_instances.clj:44]: why PKs?

File: src/metabase/models/permissions.clj (hunk 1 above)
Comment (qnkhuat) [reply src/metabase/models/permissions.clj:1039]: yea good point, we should have these insert methods in model ns to return the instances.

Comment (qnkhuat) [reply src/metabase/models/permissions.clj]: changed in abbea87184

File: src/metabase/sync/sync_metadata/fields/sync_instances.clj (hunk 2 above)
Comment (qnkhuat) [reply src/metabase/sync/sync_metadata/fields/sync_instances.clj:43]: `insert-many` == `insert-returning-pks` and we do use the result of this cal.

sorry I originally put in the description that `insert-many` == `insert-returning-pks`

File: src/metabase/models/permissions.clj [hunk 3]
```diff
@@ -1037,10 +1036,10 @@
    (t2/delete! Permissions :group_id (u/the-id group-or-id) :object [:like "/query/%"])
    (t2/delete! Permissions :group_id (u/the-id group-or-id) :object [:like "/data/%"])
    (try
-     (db/insert-many! Permissions
-       (map (fn [path-object]
-              {:group_id (u/the-id group-or-id) :object path-object})
-            (distinct (conj (->v2-path path) path))))
+     (t2/insert-returning-instances! Permissions
```
[the remark this answers is not in the document]
Comment (calherries) [reply src/metabase/models/permissions.clj:1039]: This could just be `t2/insert!`, right?

**Review by calherries — APPROVED** [review]
LGTM but I think you can save an extra db call with `grant-permissions!`

**Comment by calherries** [comment]
I just saw you used `insert-returning-instances!` because of [this comment](https://github.com/metabase/metabase/pull/29510/files/20b65259836da49cc81b4c24afb583d6a0bafdd5#r1150385995). I still don't understand the reasoning behind the change though. Can you explain?

**Comment by qnkhuat** [comment]
@calherries see https://github.com/metabase/metabase/pull/29389#issue-1633729921
> One thing I want to make explicit is that I think methods in the model namespace do something like save-task-history!, save-activiy!, and should return the inserted object by default. we might not need it, but let's make it a general convention so we don't have to think about it when using it.

**Pull request merged**